docs/en/remote-flow.rst

```diff
@@ -610,19 +657,21 @@ When the Wallet sends a response using ``direct_post.jwt`` to the Relying Party,
 
   {
     "error": "invalid_request",
-    "error_description": "The following Credentials/VP are malformed: [CredentialX, vp_token[2]]"
+    "error_description": "The vp_token is malformed, missing required parameters or incorrectly formatted"
   }
 
 Redirect URI
 ------------
 
-As defined in Section 7.2. (Response Mode "direct_post")  of the OpenID4VP specification, if the Response URI has successfully been processed, the Relying Party MUST respond with an HTTP status code of 200 and a JSON object in the response body containing the ``redirect_uri`` parameter. The Content-Type of the response SHOULD be set to ``application/json``.
+As defined in Section 7.2. (Response Mode "direct_post") of the OpenID4VP specification, if the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with an HTTP status code of 200 with ``Content-Type`` of ``application/json`` and a JSON object in the response body. If the User has been correctly authenticated, the Relying Party MUST add the ``redirect_uri`` parameter to the JSON object in the response body. 
 
-When the Relying Party provides the redirect URI, the Wallet Instance MUST redirect the user-agent to the location defined in the redirect URI.
-The redirect URI allows the Relying Party to continue the interaction with the End-User on the device where the Wallet Instance resides after the Wallet Instance has sent the Authorization Response to the response URI.
+.. note::
+    In the **Cross Device Flow** the Relying Party will pass the ``redirect_uri`` to the user-agent, whereas in the **Same Device Flow** it may also pass the ``redirect_uri`` to the Wallet Instance. When the ``redirect_uri`` is passed throught the user-agent and the response sent to the Wallet Instance does not contain the ``redirect_uri`` parameter, the Wallet Instance is not required to perform any further step.
```

Comment:
```suggestion
    In the **Cross Device Flow** the Relying Party uses the ``redirect_uri`` parameter to the user-agent, whereas in the **Same Device Flow** it may also pass the ``redirect_uri`` to the Wallet Instance. When the ``redirect_uri`` is passed through the user-agent and the response sent to the Wallet Instance does not contain the ``redirect_uri`` parameter, the Wallet Instance is not required to perform any further step.
```